src/NuGet.Core/NuGet.CommandLine.XPlat/NuGet.CommandLine.XPlat.csproj

```diff
@@ -20,23 +20,13 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.Extensions.CommandLineUtils.Sources" PrivateAssets="All" />
     <PackageReference Include="System.Diagnostics.Debug" />
+    <PackageReference Include="Microsoft.Build" ExcludeAssets="runtime" />
   </ItemGroup>
 
-  <!-- Microsoft.Build.Locator is needed when debugging, but should not be used in the assemblies we insert.
-       It also generates a build error if the output directory contains MSBuild dlls, so exclude msbuild assets when using the locator. -->
-  <Choose>
-    <When Condition=" '$(Configuration)' == 'Debug' ">
-      <ItemGroup>
-        <PackageReference Include="Microsoft.Build" ExcludeAssets="runtime" />
-        <PackageReference Include="Microsoft.Build.Locator" />
-      </ItemGroup>
-    </When>
-    <Otherwise>
-      <ItemGroup>
-        <PackageReference Include="Microsoft.Build" />
-      </ItemGroup>
-    </Otherwise>
-  </Choose>
+  <!-- Microsoft.Build.Locator is only used when debugging, and the compiler will trim this dependency from Release assemblies we insert. -->
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Build.Locator"  />
```

Comment:
I wonder if we need `PrivateAssets="All"` so that the `.nupkg` from this project doesn't pull `Microsoft.Build.Locator` into downstream dependencies?  Can you check the package you built locally and see if this shows up in the .nupsec/.nupkg?

Reply:
Good point - done.
Before: 
![image](https://user-images.githubusercontent.com/49205731/184990500-8be706f9-83a4-46a6-93ac-9feef0a3f387.png)

After:
![image](https://user-images.githubusercontent.com/49205731/184990524-6c3aa099-3f35-4cbc-81da-8dac7b8e6ca9.png)